akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -311,7 +311,16 @@ private[stream] class VirtualSubscriber[T](val owner: VirtualPublisher[T]) exten
  */
 private[stream] class VirtualPublisher[T]() extends Publisher[T] {
   @volatile var realPublisher: Publisher[T] = null
-  override def subscribe(s: Subscriber[_ >: T]): Unit = realPublisher.subscribe(s)
+  override def subscribe(s: Subscriber[_ >: T]): Unit = {
+    val sub = realPublisher.subscribe(s)
+    realPublisher match {
+      case _: SupportingMultipleSubscribers   ⇒
```

Comment:
@rkuhn I would like that we decide what to do here, so I can merge.

The following tests fail if I remove the SupportingMultipleSubscribers marker:

* Failed Source must emit error immediately
* Empty Source must complete immediately
* Singleton Source must produce elements to later subscriber
* A Flow based on a Future must produce elements with multiple subscribers
* A Flow based on a Future must produce elements to later subscriber
* A Flow based on a Future must allow cancel before receiving element

If we remove the marker I have to remove or rewrite those tests.